Tests/GRPCTests/HTTP1ToGRPCServerCodecTests.swift

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2019, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Foundation
+import XCTest
+import NIO
+import NIOHTTP1
+@testable import GRPC
+import EchoModel
+import EchoImplementation
+import Logging
+
+
+class HTTP1ToGRPCServerCodecTests: GRPCTestCase {
+  var channel: EmbeddedChannel!
+
+  override func setUp() {
+    super.setUp()
+    let logger = Logger(label: "io.grpc.testing")
+    let handler = HTTP1ToGRPCServerCodec<Echo_EchoRequest, Echo_EchoResponse>(logger: logger)
+    self.channel = EmbeddedChannel(handler: handler)
+  }
+
+  override func tearDown() {
+    super.tearDown()
+    XCTAssertNoThrow(try self.channel.finish())
+  }
+
+  func makeRequestHead() -> HTTPRequestHead {
+    return HTTPRequestHead(
+      version: .init(major: 2, minor: 0),
+      method: .POST,
+      uri: "/echo.Echo/Get"
+      )
+  }
+
+  func testDeserializationErrorOnInvalidMessageBytes() throws {
+    XCTAssertNoThrow(try self.channel.writeInbound(HTTPServerRequestPart.head(self.makeRequestHead())))
+    var buffer = self.channel.allocator.buffer(capacity: 0)
+    buffer.writeInteger(UInt8(0))  // not compressed
+    buffer.writeInteger(UInt32(3)) // message is 3 bytes
+    buffer.writeBytes([42, 42, 42])
+    XCTAssertThrowsError(try self.channel.writeInbound(HTTPServerRequestPart.body(buffer))) { error in
+      let withContext = error as? GRPCError.WithContext
+      XCTAssertTrue(withContext?.error is GRPCError.DeserializationFailure)
+      XCTAssertEqual(withContext?.error.makeGRPCStatus().code, .internalError)
```

Comment:
nit: I prefer to put the expectation first (also elsewhere I guess) — that's what I learned at Google. I don't want to force my will onto the entire project, though.

Reply:
I do it this way for practical reasons: the type inference comes from the first arg in `XCTAssertEqual`.

There are two useful aspects to this:
- if the left hand side is non-optional, then the right hand side must be also be non-options. If the left hand side is optional, the right hand side doesn't have to be optional.
- the type inference saves some typing; i.e. we can have `.internalError` instead of `GRPCStatus.Code.internalError`

Reply:
I could have sworn that I had both of these in my tests with the other order as well, are you sure?

E.g. I thought had written tests like this in the past:

```
XCTAssertEqual(0, optionalInt)
XCTAssertEqual(.someCase, someEnum)
```

Reply:
Not so sure now, no. Seems to work. I definitely hit this problem in the past, maybe there was a bug before 🤷‍♂️